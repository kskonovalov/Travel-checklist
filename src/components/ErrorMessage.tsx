import React from 'react';
import styled from 'styled-components';

const Message = styled.div`
  text-align: center;
  color: #932727;
  font-size: 12px;
  margin-top: 10px;
`;

const ErrorMessage: React.FC<{ text: string }> = ({ text }) => {
  return <Message>{text}</Message>;
};

export default ErrorMessage;
