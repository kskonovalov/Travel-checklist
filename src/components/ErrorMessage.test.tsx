import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import ErrorMessage from './ErrorMessage';

configure({ adapter: new Adapter() });

const setUp = ({ text }: { text: string }) => {
  return <ErrorMessage text={text} />;
};

describe('Component: ErrorMessage', () => {
  test('Should match snapshot', () => {
    const text = 'Test text';
    const component = shallow(setUp({ text }));
    expect(component).toMatchSnapshot();
  });

  test('Should render correct text from param', () => {
    const text = 'Test text';
    const component = shallow(setUp({ text }));
    expect(component.text()).toEqual(text);
  });
});
