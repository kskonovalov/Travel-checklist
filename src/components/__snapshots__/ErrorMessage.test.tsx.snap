// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ErrorMessage Should match snapshot 1`] = `
<styled.div>
  Test text
</styled.div>
`;
